js/ci-configs/.gitlab-ci.yml

```diff
@@ -0,0 +1,79 @@
+image: node:8
+
+# Cache dependencies.
+cache:
+  paths:
+    - node_modules/
+
+stages:
+  - test
+  - lint
+  - deploy-gl-pages
+  - deploy-heroku
+
+# Define the desired stages.
+test:
+  stage: test
+  script:
+    - yarn install
+    - yarn test
+
+  # https://gitlab.com/gitlab-org/gitlab-ce/issues/29642
+  coverage: '/All files[^|]*\|[^|]*\s+([\d\.]+)/'
+
+lint:
+  stage: lint
+  script:
+    - yarn install
+    - yarn lint
+
+# Code quality setup.
+code_quality:
+  stage: test
+  image: docker:stable
+  variables:
+    DOCKER_DRIVER: overlay2
+  allow_failure: true
+  services:
+    - docker:stable-dind
+  script:
+    - export SP_VERSION=$(echo "$CI_SERVER_VERSION" | sed 's/^\([0-9]*\)\.\([0-9]*\).*/\1-\2-stable/')
+    - docker run
+      --env SOURCE_CODE="$PWD"
+      --volume "$PWD":/code
+      --volume /var/run/docker.sock:/var/run/docker.sock
+      "registry.gitlab.com/gitlab-org/security-products/codequality:$SP_VERSION" /code
+  artifacts:
+    reports:
+      codequality: gl-code-quality-report.json
+
+# Auto-deployment to GitLab pages.
+pages:
+  stage: deploy-gl-pages
+  script:
+    - (if [[ -z $DEPLOY_TO_GL_PAGES ]]; then echo "$DEPLOY_TO_GL_PAGES is unset. Set this value to true if you'd like to enable auto-deployment to GitLab pages."; exit 0; fi);
+    - (if [[ $DEPLOY_TO_GL_PAGES = false ]]; then echo "Skipping auto-deployment to GitLab pages."; exit 0; fi);
```

Comment:
Is there a way to check falsity and unset variables in one step? It was confusing why this step needs 2 checks and the other only 1.

Reply:
This differs from the `deploy-heroku` step as it's a boolean value, therefore it can be either unset or false.

Your supposition is right, but not sure if Bash provides a "one-line way" similarly to JavaScript.

---

Tried to read up on the topic, but couldn't find an existing solution.

- Here's an [**answer**](https://stackoverflow.com/a/21210966/9599137) that demonstrates the right ways of checking boolean values in Bash. It's seemingly not that easy as in JS.
- To test whether a variable is set or unset, the `-z` flag should be used.

However, the combination of the two doesn't really exist.

Reply:
Another way could be to use [**only/expect**](https://docs.gitlab.com/ee/ci/yaml/#onlyvariablesexceptvariables) in the case of variables, but it'd raise further questions - e.g. how to `echo` the message?

Reply:
> This differs from the deploy-heroku step as it's a boolean value

Yep, too bad it's this complicated.

> Another option could be to use only/expect in the case of variables

That's actually a great find. Let's use that instead. I think it's very straightforward looking at the config and doesn't require the console messages.

Reply:
Check 195d56a. :)

Created an [**experimental branch**](https://gitlab.com/c-hive/poe-sniper-electron/merge_requests/30) in the other repo - the only difference is that I directly left the `master` branch restriction for the sake of completeness.

Works as expected, the flow is much clearer.